### PR TITLE
Add metrics for completing/aborting a merge after conflicts

### LINKED ISSUE #5394: [Metrics] Merge conflict resolution
With the general goal of reducing friction for people collaborating on software, we want to instrument the app to understand the following:

- [ ] Given that an attempted merge results in a merge conflict, what percentage of those eventually result in a successful merge?
- [ ] Of all attempted merges, what percentage result in a successful merge?

I think we have attempted merges that result in conflicts, so the missing piece here is to be able to understand whether those conflicts eventually get resolved successfully or whether they're just abandoned. A thing we've heard in user research is people saying some version of, "and then I just had to start over because I couldn't resolve the conflict." We obviously want to avoid this, so understanding how common it is when people do get into conflicts will be helpful in us figuring out whether we're making progress here.

The ideal scenario is that anytime a person attempts to merge code from one branch to another, that they're able to do so successfully.

Tasks
- [x] Create issue in analytics repo: https://github.com/github/analytics/issues/712
- [x] Add metrics to app: #5810 
- [x] Update metrics documentation #5879 
- [x] Update marketing site - https://github.com/github/desktop.github.com/pull/119
- [x] Update central - https://github.com/github/central/pull/411
- [ ] 🚢 

cc: @telliott27

### PR DESCRIPTION
**Pull request by iAmWillShepherd**
⚠️ This PR has associated work: https://github.com/desktop/desktop/issues/5394#issue-350225825 ⚠️ 

Fixes #5394 

## How to test
1. Create a repo and produce some conflicts
1. Open that repo in Desktop (you should be notified that conflicts exist
1. Note the values of `mergeAbortedAfterConflictsCount ` and  `mergeSuccessAfterConflictsCount` from the database as _n_ and _m_ respectfully. 

### Abort
1. Open terminal and execute `git merge --abort`
1. Check the database: `mergeAbortedAfterConflictsCount ` should be **n + 1**

### Complete
1. Fix the conflicts
1. Make a commit
1. Check the database: `mergeSuccessAfterConflictsCount` should be **m + 1**

### CODE REVIEW

**Review by outofambit — CHANGES_REQUESTED** [review]
thanks for taking this the last mile, @iAmWillShepherd! a few blocking requests.

how can we easily test that this works?

File: app/src/lib/stats/stats-store.ts
Comment (iAmWillShepherd) [inline app/src/lib/stats/stats-store.ts]: Need to append _Count_ to these new measures.

File: app/src/lib/app-state.ts
Comment (iAmWillShepherd) [reply app/src/lib/app-state.ts]: This interface only exists to be used within another interface. I contemplated not exporting it just in case we wanted to do things like `let a: IConflictState | null = null`. Perhaps it's best not to assume we will need this and instead wait for the need to export this. I'll go ahead and add documentation, but it will be redundant as this is documented in the parent interface.

File: app/src/lib/stores/app-store.ts
Comment (iAmWillShepherd) [reply app/src/lib/stores/app-store.ts]: Many things that could happen to cause this, none of which we actually care about here. Distinguishing between them makes no difference at the moment because having a branch is an invariant in this function.

Comment (iAmWillShepherd) [reply app/src/lib/stores/app-store.ts]: I probably should add that _something_ could be good or bad, in either case, we don't care since having a branch selected is the thing we care about.

File: app/src/lib/stores/app-store.ts
Comment (iAmWillShepherd) [reply app/src/lib/stores/app-store.ts]: If we put this statement on line 1491, we'd actually potentially increase the amount of work we have to do.  The check on line 1496 ensures we don't do extra work and the check on 1501 can return from the function before getting to this check.

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [reply app/src/lib/stores/app-store.ts]: We chatted and agreed to make a named variable here for clarity :)

**Review by outofambit — DISMISSED** [review]
tested this out locally and i think we're good 👍

File: app/src/lib/dispatcher/error-handlers.ts
Comment (shiftkey) [inline app/src/lib/dispatcher/error-handlers.ts]: Is there a reason why we need to make these calls alongside existing conflict-related calls? Could `Dispatcher.recordMergeConflictFromPull` and `Dispatcher. recordMergeConflictFromExplicitMerge` also invoke `this.appStore._mergeConflictDetected()` internally?

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [reply app/src/lib/stores/app-store.ts]: > We chatted and agreed to make a named variable here for clarity :)

Did this happen?

**Review by shiftkey — CHANGES_REQUESTED** [review]
In case this gets merged before #5810 is clarified I've added some further feedback and assigned myself to it.

File: app/src/lib/stores/app-store.ts
Comment (iAmWillShepherd) [reply app/src/lib/stores/app-store.ts]: It will happen 😉

**Comment by kenia-borges** [comment]
I can not do this, it goes beyond my limits.

Obter o Outlook para iOS<https://aka.ms/o0ukef>
________________________________

File: app/src/lib/app-state.ts
Comment (shiftkey) [reply app/src/lib/app-state.ts]: 👍 to writing some better docs around this. I'll have a shot at a PR for this later and you can sanity check me.

My gut feeling for moving it under `IChangesState` is that we are starting to do more work in the Changes tab around identifying which files are conflicted and when those conflicts have been resolved, and keeping those two pieces of state in sync is probably easier when they're in the same area of the object graph.

Comment (iAmWillShepherd) [reply app/src/lib/app-state.ts]: @shiftkey, I addressed this request in https://github.com/desktop/desktop/pull/5901.

File: app/src/lib/dispatcher/error-handlers.ts
Comment (iAmWillShepherd) [reply app/src/lib/dispatcher/error-handlers.ts]: I can't really speak on calling `dispatcher. recordMergeConflictFromExplicitMerge()`, but I can explain the reasoning for putting the call `dispatcher.mergeConflictDetected()` in both cases: the method names `recordMergeConflictFromPull` and `recordMergeConflictFromExplicitMerge` doesn't hint that it should do anything else other than increment our stats counters.

**Comment by iAmWillShepherd** [comment]
🍏

File: app/src/lib/dispatcher/error-handlers.ts
Comment (shiftkey) [reply app/src/lib/dispatcher/error-handlers.ts]: > doesn't hint that it should do anything else other than increment our stats counters

Should they? I think that's the part I was misunderstanding, so thanks for clarifying.

Thinking more on this, this feels like a chance to encapsulate related behaviour and make the dispatcher reflect how we want to use it, rather than following the old pattern of it being 1-1 with `AppStore`.

We could end up with a more readable flow, like this:

```ts
switch (command) {
  case 'pull':
    dispatcher.mergeConflictDetectedFromPull()
    break
  case 'merge':
    dispatcher.mergeConflictDetectedFromExplicitMerge()
    break
}
```

The dispatcher functions could then look like this:

```
/**
 * Updates the application state to indicate a conflict is in-progress,
 * and increments the relevant metric.
 */
public mergeConflictDetectedFromExplicitMerge() {
  this.appStore._mergeConflictDetected()
  return this.statsStore.recordMergeConflictFromExplicitMerge()
}
```

Thoughts?

Comment (iAmWillShepherd) [reply app/src/lib/dispatcher/error-handlers.ts]: I like it. It's definitely more descriptive of what we're doing here. #ReduxStateOfMind! 😉

**Comment by iAmWillShepherd** [comment]
🍏

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [inline app/src/lib/stores/app-store.ts]: :art: i don't think we need this comment

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [inline app/src/lib/stores/app-store.ts]: :art: i don't think we need this comment

**Review by shiftkey — COMMENTED** [review]
Just a couple of suggestions

**Comment by iAmWillShepherd** [comment]
💯

**Review by outofambit — APPROVED** [review]

File: app/src/lib/stores/app-store.ts
Comment (shiftkey) [inline app/src/lib/stores/app-store.ts]: :art: `const { repository } = selection`

**Review by shiftkey — APPROVED** [review]
One suggestion to cleanup, but I'll let @outofambit merge this at their leisure so we can wrap this up :tada:

**Comment by outofambit** [comment]
thanks @shiftkey! 🚀

File: app/src/lib/stores/app-store.ts
Comment (outofambit) [inline app/src/lib/stores/app-store.ts]:
```suggestion
    const { repository } = selection
```

**Review by ghost — COMMENTED** [review]
#14275

**Comment by chinagourmet888** [comment]
git merge --abort

File: app/src/lib/dispatcher/dispatcher.ts
Comment (pentorwrt-ship-it) [inline app/src/lib/dispatcher/dispatcher.ts]: Imageem<
E. G., อยู่ IP มีลักษณะอย่างไร
ที่อยู่ IP จะเป็นตัวระบุ เช่น 203.0.113.42 หรือ 2001:0002:14:5:1:2:bf35:2610

หน้านี้อาจมีเนื้อหาที่แปลโดยใช้เทคโนโลยี AI การแปลโดย TO: Kenneth Reed
Datawatch Systems
4520 East-West Highway, Suite 200
Bethesda, MD 20814
FROM: Vanessa Arias, on behalf of the FIPS 201 Evaluation Program
DATE: 04/10/2025
RE: Approval of Products/Services for Listing on the GSA APL
Office of Management and Budget (OMB) issued M-19-17, which requires Federal
Agencies to procure only qualified products and services that are compliant with OMB
policy, NIST standards, and supporting technical specifications, such as those listed on
the GSA Approved Products List (APL) when implementing HSPD-12 and ICAM into
their environment. Procuring approved products and services facilitates the
government-wide objective of a federated and interoperable ICAM segment architecture.
It ensures compliance, consistency, and alignment of commercially available products
and services with the requirements and functional needs of government ICAM
implementers.
Accordingly, the FIPS 201 Evaluation Program (Program) is delighted to inform you that
the following product/service has passed testing against all applicable Testing
Requirements. Please see the attached document that summarizes the specific parameters
of the approval (e.g., product/service profile and what your product/service was tested
with). Your product/service will now be listed on the APL, as shown below. If there are
any errors in the listing information, please notify us within five (5) business days of
receiving this letter.
Docusign Envelope ID: 2B50491C-57ED-4472-A45C-9AD9E11E1383 1 Category- 13.01 Topology Cloud PACS Infrastructure (PI)
1.1 APL Certification Number
● APL# 10117
o DataWatch Systems Site Controller
o DataWatch Access Control Panel
o DataWatch PACS Client Software
1.2 Product/Service Information
Organization Name: DataWatch System
Name of Product or Service: DW – Site Controller
Part Number: SC3.6
Hardware Version: N/A
Software Version: V3.31
Firmware Version: N/A
Organization Name: DataWatch System
Name of Product or Service: Access Control Panel
Part Number: E400M
Hardware Version: E400M
Software Version: N/A
Firmware Version: 07/01/2024@10:02
Organization Name: DataWatch System
Name of Product or Service: PACS Client Software
Part Number: D3000 Express
Hardware Version: N/A
Software Version: 2.7.0.1
Firmware Version: N/A
1.3 Document History
Status Date Comments
Approved 10/15/2019 Approval date for APL#10117
Updated 04/09/2025 Update of APL # 10117.
Updated 04/10/2025 Updated the address for Datawatch.
1.4 Restrictions:
Datawatch systems Site Controller (APL# 10117) and HID Validation Infrastructure for
Datawatch (APL# 10118) have been tested and approved as fully compliant FICAM
solution components. The agency deploying the solution is responsible for verifying that
the deployment architecture (e.g., on-site, private cloud, public cloud, etc.) meets the
Docusign Envelope ID: 2B50491C-57ED-4472-A45C-9AD9E11E1383 agency's security requirements, such as FedRAMP. The end-to-end components used to
test the FICAM compliance of your approved solution are listed below.
● Validation System (APL# 10118)
o pivCLASS Registration Engine (Datawatch Systems)
o pivCLASS Certificate Manager
o pivCLASS Reader Services
o pivCLASS Authentication Module
● PIV Reader
o pivCLASS RP40 Contactless Reader (APL# 10003)
o pivCLASS RKCLB40 Contact/Contactless + PIN + BIO (APL# 10052)
Approval of the above product/service indicates your cooperation with the Program and
successful testing against a rigorous, comprehensive set of functional requirements
derived from government-wide specifications. Testing included positive and negative test
cases and threat vectors.
Given the strategic government-wide importance of the APL, the scope and extent of
testing performed on your product/service, and the rigor of that testing, the Program
would like to acknowledge the noteworthiness of the approval and listing. Your
product/service will now be available for procurement by all federal agencies.
Please note that continued listing on the APL requires ongoing conformance to all
applicable requirements, including requirements added or amended over time. A listed
product/service can become non-conformant for various reasons, including but not
limited to:
● New Functional Requirements and Test Cases—The Functional Requirement and
Test Case (FRTC) documents are effective immediately. All approved Physical
Access Control System (PACS) solutions must pass testing against the new and
revised FRTC requirements before the effective date of the new/revised
requirements.
● Specific problems are discovered in a vendor’s product or services (or class of
products/services) after being listed on the APL. The affected vendor(s) are
notified that the identified product(s) and services(s) must be improved within the
time frames specified in the FRTC, commensurate with the severity level of the
problem.
● Updates to testing requirements due to concerns provided to the Evaluation
Program by Federal agency buyers.
● Updates to testing requirements due to vendor-provided issues to the Evaluation
Program.
● A product whose last certification letter is older than 12 months and has yet to
receive an APL Extension approval.
Docusign Envelope ID: 2B50491C-57ED-4472-A45C-9AD9E11E1383 Refer to the applicable Agreements and Program documents for details regarding
ongoing compliance, including severity levels and remediation time frames. Products and
services listed as nonconforming will be removed from the APL and added to the
Removed Products List (RPL).
If you have any questions, please feel free to contact me.
FIPS 201 Evaluation Program Manager
Vanessa Arias 
GSA OGP
(240) 246-5887
Vanessa.Arias@gsa.gov
Docusign Envelope ID: 2B50491C-57ED-4472-A45C-9AD9E11E1383
4/21/2025

File: app/src/lib/dispatcher/error-handlers.ts
Comment (pentorwrt-ship-it) [inline app/src/lib/dispatcher/error-handlers.ts]: Fixes #22423

Comment (pentorwrt-ship-it) [inline app/src/lib/dispatcher/error-handlers.ts]: E. G., MOVELOG oem logon emotional gtr root container mea pae

**Pull request merged**
